.circleci/config.yml

```diff
@@ -31,6 +31,7 @@ store-log-artifacts: &store-log-artifacts
 flow-type-key: &flow-typed-cache-key flow-typed-{{ checksum "yarn.lock" }}-5
 npm-cache-key: &npm-cache-key node-v10-{{ checksum "yarn.lock" }}-5
 bundle-cache-key: &bundle-cache-key v2-bundler-gems-{{ .Environment.DB }}-{{ arch }}-{{ .Branch }}-{{ checksum "Gemfile.lock" }}
+assets-cache-key: &assets-cache-key v1-asset-cache-{{ .Branch }}-{{ .Revision }}
```

Comment:
In [documentation example](https://circleci.com/docs/2.0/caching/#full-example-of-saving-and-restoring-cache) `arch` is also included in cache key. I see no reason for this given that assets should be arch independent. Let me know if I'm missing something.

The idea is to cache not more often than per revision tested. Ideally we can figure out from sources whether anything changed but it seems too complicated. Most likely need to make sure that `yarn.lock` and `Gemfile.lock` didn't change as well no other js file in our repo.

Still doable but maybe not worth the effort. WDYT?

Reply:
Well doing so does not really help much, does it? You are unlikely to run the same tests more than once for each revision ... Even for different type of DB, or you would wait for assets precompilation first and cache it for other pipelines can reuse it... But that means you will not gain anything for it, since you add 2' overhead before been able to run any pipeline instead of having the pipelines run in parallel

Reply:
Yeah, we will need to compute for every workflow.

Reply:
But it is about sharing between revisions. If we compute some checksum to know whether something is changing, we will not use `revision` but share across any revisions.

Reply:
Yes but how likely are you going to test the same revision more than once?

Reply:
Also you can completely base your cache key on `app/assets` and `app/javascript` and other directories
The issue is when it comes to vendored assets ...

Reply:
I think many pull requests do not touch assets so it would be often the case when cache is valid. Also even partial cache would have some usefulness. But anyway, I agree that unlikely the benefit will be worth the additional complexity. That's why I left it like it is.

Reply:
Well, in the contrary many PR touch assets :)
Look at how many PR @josemigallas or @damianpm are doing, well in general the @3scale/system-ux is doing many many PR here.